Survey/Survey/Sources/Presentation/Modules/Entry/EntryViewModel.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  EntryViewModel.swift
+//  Survey
+//
+//  Created by Taher on 2/11/22.
+//
+
+import Combine
+import SwiftUI
+
+class EntryViewModel: ObservableObject {
+
+    @Published var viewType: EntryViewType = .splash
+    var didFinishSplashAnimation = PassthroughSubject<Void, Never>()
+    let coordinator: AppCoordinatorViewModel
+    private var disposables = Set<AnyCancellable>()
+
+    init(coordinator: AppCoordinatorViewModel) {
+        self.coordinator = coordinator
+
+        didFinishSplashAnimation.sink { [weak self] _ in
+            self?.goToNextView()
+        }
+        .store(in: &disposables)
+    }
+
+    private func goToNextView() {
+        viewType = .login
+        // To-Do: add logic to check authentication and route accordingly
```

Comment:
```suggestion
        // TODO: add logic to check authentication and route accordingly
```

Reply:
update in this [commit](https://github.com/Shayokh144/ic-survey-swiftui-combine/pull/41/commits/f10d87bdfe1267e19fe24bb94768bd66860444a7)